go/consensus/tendermint/api/context.go

```diff
@@ -94,30 +85,56 @@ func FromCtx(ctx context.Context) *Context {
 	return abciCtx
 }
 
+// NewContext creates a new context.
+func NewContext(
+	ctx context.Context,
+	mode ContextMode,
+	currentTime time.Time,
+	gasAccountant GasAccountant,
+	appState ApplicationState,
+	state mkvs.Tree,
+	blockHeight int64,
+	blockCtx *BlockContext,
+) *Context {
+	c := &Context{
+		mode:          mode,
+		currentTime:   currentTime,
+		gasAccountant: gasAccountant,
+		appState:      appState,
+		state:         state,
+		blockHeight:   blockHeight,
+		blockCtx:      blockCtx,
+		logger:        logging.GetLogger("consensus/tendermint/abci").With("mode", mode),
+	}
+	c.Context = context.WithValue(ctx, contextKey{}, c)
```

Comment:
is this allowed inside the literal?

oh unless there's some logical separation to justify putting it here

Reply:
Oh you mean if we could move initializing the `Context` inside the literal that constructs `c`? Yeah the problem is that it requires a reference to `c` which doesn't exist yet at that point :-)